Parse/Internal/Commands/CommandRunner/URLSession/Session/PFURLSession.m

```diff
@@ -161,14 +167,18 @@ - (BFTask *)performFileDownloadURLRequestAsync:(NSURLRequest *)request
 }
 
 - (BFTask *)_performDataTask:(NSURLSessionDataTask *)dataTask withDelegate:(PFURLSessionDataTaskDelegate *)delegate {
+    [self.delegate urlSession:self willPerformURLRequest:dataTask.originalRequest];
+
     @weakify(self);
     return [BFTask taskFromExecutor:[BFExecutor defaultExecutor] withBlock:^id{
         @strongify(self);
         NSNumber *taskIdentifier = @(dataTask.taskIdentifier);
         [self setDelegate:delegate forDataTask:dataTask];
 
-        BFTask *resultTask = [delegate.resultTask continueWithBlock:^id(BFTask *task) {
+        BFTask *resultTask = [delegate.resultTask continueWithExecutor:[BFExecutor defaultPriorityBackgroundExecutor] withBlock:^id(BFTask *task) {
```

Comment:
Changed the executor - make sure the delegate callback is made on asynchronous thread and never inline (default executor can execute immediately in 19/20 chances).
`continueAsyncWithBlock` - good point.